### PR TITLE
NOT_AN_OBJECT: throw for null; add additional tests

### PR DESCRIPTION
**Pull request by alxndrsn**
Also separates test cases so specific failures can be understood.

Without the fix, tests fail:

```
  test.basics.js-http
    Should error when document is not an object #0
      ✔ should error for .post()
Testing on: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) HeadlessChrome/123.0.6312.4 Safari/537.36
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #1
      1) should error for .post()
      2) should error for .put()
      3) should error for .bulkDocs()
    Should error when document is not an object #2
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #3
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #4
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #5
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()

  test.basics.js-local
    Should error when document is not an object #0
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #1
      4) should error for .post()
      5) should error for .put()
      6) should error for .bulkDocs()
    Should error when document is not an object #2
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #3
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #4
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()
    Should error when document is not an object #5
      ✔ should error for .post()
      ✔ should error for .put()
      ✔ should error for .bulkDocs()


  30 passing (515ms)
  6 failing

  1) test.basics.js-http
       Should error when document is not an object #1
         should error for .post():

      AssertionError: expected 'Cannot read properties of null (reading \'_id\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_id')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)

  2) test.basics.js-http
       Should error when document is not an object #1
         should error for .put():

      AssertionError: expected 'Cannot read properties of null (reading \'_id\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_id')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)

  3) test.basics.js-http
       Should error when document is not an object #1
         should error for .bulkDocs():

      AssertionError: expected 'Cannot read properties of null (reading \'_attachments\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_attachments')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)

  4) test.basics.js-local
       Should error when document is not an object #1
         should error for .post():

      AssertionError: expected 'Cannot read properties of null (reading \'_id\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_id')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)

  5) test.basics.js-local
       Should error when document is not an object #1
         should error for .put():

      AssertionError: expected 'Cannot read properties of null (reading \'_id\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_id')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)

  6) test.basics.js-local
       Should error when document is not an object #1
         should error for .bulkDocs():

      AssertionError: expected 'Cannot read properties of null (reading \'_attachments\')' to equal 'Document must be a JSON object'
      + expected - actual

      -Cannot read properties of null (reading '_attachments')
      +Document must be a JSON object
      
      at Context.<anonymous> (test.basics.js:767:32)
```

### CODE REVIEW

File: packages/node_modules/pouchdb-core/src/adapter.js [hunk 1]
```diff
@@ -197,7 +197,7 @@ class AbstractPouchDB extends EventEmitter {
         callback = opts;
         opts = {};
       }
-      if (typeof doc !== 'object' || Array.isArray(doc)) {
+      if (doc == null || typeof doc !== 'object' || Array.isArray(doc)) {
```
Comment (SourceR85) [inline packages/node_modules/pouchdb-core/src/adapter.js:200]: Checking `null` non-strict will match every falsy value, you can reduce it to just `!doc`
```suggestion
      if (!doc || typeof doc !== 'object' || Array.isArray(doc)) {
```

Alt.:
```suggestion
      if (doc === null || typeof doc !== 'object' || Array.isArray(doc)) {
```

File: packages/node_modules/pouchdb-core/src/adapter.js [hunk 2]
```diff
@@ -208,7 +208,7 @@ class AbstractPouchDB extends EventEmitter {
         cb = opts;
         opts = {};
       }
-      if (typeof doc !== 'object' || Array.isArray(doc)) {
+      if (doc == null || typeof doc !== 'object' || Array.isArray(doc)) {
```
Comment (SourceR85) [inline packages/node_modules/pouchdb-core/src/adapter.js:211]: As above

File: packages/node_modules/pouchdb-core/src/adapter.js [hunk 3]
```diff
@@ -778,7 +778,7 @@ class AbstractPouchDB extends EventEmitter {
       }
 
       for (var i = 0; i < req.docs.length; ++i) {
-        if (typeof req.docs[i] !== 'object' || Array.isArray(req.docs[i])) {
+        if (req.docs[i] == null || typeof req.docs[i] !== 'object' || Array.isArray(req.docs[i])) {
```
Comment (SourceR85) [inline packages/node_modules/pouchdb-core/src/adapter.js:781]: Same here

**Review by SourceR85 — CHANGES_REQUESTED** [review]

File: packages/node_modules/pouchdb-core/src/adapter.js (hunk 1 above)
Comment (alxndrsn) [reply packages/node_modules/pouchdb-core/src/adapter.js:200]: > Checking `null` non-strictly will match every _falsy_ value, you can reduce it to just `!doc`

Checking `null` non-strictly will match `undefined` and `null` only.

```js
> [0,'',false,null,undefined].map(it => it == null)
[ false, false, false, true, true ]
```

Comment (alxndrsn) [reply packages/node_modules/pouchdb-core/src/adapter.js:200]: Either suggestion will work though :+1:

**Review by SourceR85 — APPROVED** [review]
LGTM

**Pull request merged**